certbot-nginx/certbot_nginx/obj.py

```diff
@@ -96,6 +96,11 @@ def __repr__(self):
     def super_eq(self, other):
         """Check ip/port equality, with IPv6 support.
         """
+        # Check if both objects have wildcard equivalent addresses.
+        # If so, we're done.
+        any_addresses = ('0.0.0.0', '*', '')
+        if self.tup[0] in any_addresses and other.tup[0] in any_addresses:
+            return True
```

Comment:
Oh true, duh. I'll fix the port checking at least and add a unit test. Maybe add v6 checking in a follow-up?